src/cephadm/cephadmlib/file_utils.py

```diff
@@ -64,11 +64,13 @@ def populate_files(config_dir, config_files, uid, gid):
 
 
 def touch(
-    file_path: str, uid: Optional[int] = None, gid: Optional[int] = None
+    file_path: str, uid: Optional[int] = None, gid: Optional[int] = None, mode: Optional[int] = None
 ) -> None:
     Path(file_path).touch()
     if uid and gid:
         os.chown(file_path, uid, gid)
+    if mode:
+        os.chmod(file_path, mode)
```

Comment:
@adk3798 the reasoning for this change is that `grafana.db` requires the mode to be `472`. This worked on a normal bootstrapped cluster. but when I upgraded from `ceph-ci/ceph:main` image to the one I got from the shaman build, the permission looked a bit weird. I am not fully sure what could be the reason.
```
[root@ceph-node-00 grafana.ceph-node-00]# ll data
total 1388
-r--rwx-w-. 1 polkitd unbound 1421312 Mar  1 10:06 grafana.db
```

Reply:
this is the image I used to upgrade my cluster: `quay.ceph.io/ceph-ci/ceph:wip-nizamial09-testing-2024-03-01-1217`

Reply:
i am thinking maybe on the upgrade scenarios, we don't need to create the `grafana.db` again using the touch command because it is already there. So if its already present, how can I skip creation of this particular file? or maybe even the folder?